compiler/src/org.graalvm.compiler.lir/src/org/graalvm/compiler/lir/gen/LIRGenerator.java

```diff
@@ -449,29 +455,58 @@ public Variable emitForeignCall(ForeignCallLinkage linkage, LIRFrameState frameS
 
     @Override
     public void emitStrategySwitch(JavaConstant[] keyConstants, double[] keyProbabilities, LabelRef[] keyTargets, LabelRef defaultTarget, Variable value) {
+
         int keyCount = keyConstants.length;
         SwitchStrategy strategy = SwitchStrategy.getBestStrategy(keyProbabilities, keyConstants, keyTargets);
+
+        double minDensity = 1 / Math.sqrt(strategy.getAverageEffort() * tableSwitchBonus);
+
+        Optional<Hasher> hasher = Optional.empty();
+        if (supportsHashTableSwitch()) {
```

Comment:
Having a `supportsHashTableSwitch()` seems odd. Could we instead add a method that returns the hasher and the default implementation just returns `Optional.empty()`?